install/01_clusterautoscaler.crd.yaml

```diff
@@ -49,6 +51,10 @@ spec:
                   feature flag. Should CA ignore DaemonSet pods when calculating resource
                   utilization for scaling down. false by default
                 type: boolean
+              maxNodeProvisionTime:
+                description: Maximum time CA waits for node to be provisioned
+                pattern: ([0-9]*(\.[0-9]*)?[a-z]+)+
```

Comment:
just a question about the regex, do we need to restrict the alphanumeric on this? it seems like it would accept a wide range of inputs.

Reply:
Thanks for the prompt review @elmiko :+1: 

Re the regex, I copied it from the similar interfaces in the scaleDown section e.g delayAfterAdd etc - those support multiple formats e.g "10m" "30s" etc - from the CA docs it looks like --max-node-provision-time supports the same (they're all flag.Duration):

https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/main.go#L137

Looking at the underlying code, it uses `time.ParseDuration` which does describe the exact same regex:

https://golang.org/src/time/format.go?s=40541:40587#L1364

Reply:
cool, works for me!

Reply:
We use an alternative regex already for the MHC `time.Duration` fields that is more restrictive than your current one and should guarantee a valid duration, perhaps we could use that one instead? https://github.com/openshift/machine-api-operator/blob/27db623eabbb97e5cec9a47863da40d119b250e8/install/0000_30_machine-api-operator_07_machinehealthcheck.crd.yaml#L76

Reply:
Done - and thanks for pointing to nodeStartupTimeout - I suspect that may also need adjustment for this to work well in a baremetal env due to the time taken for reboots.